gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/RefreshTokenAuth2Provider.java

```diff
@@ -0,0 +1,231 @@
+package com.google.cloud.hadoop.fs.gcs;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageOptions;
+import com.google.cloud.hadoop.util.AccessTokenProvider;
+import com.google.cloud.hadoop.util.RedactedString;
+import com.google.common.flogger.GoogleLogger;
+import com.google.common.io.BaseEncoding;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.http.HttpHeaders;
+
+public class RefreshTokenAuth2Provider implements AccessTokenProvider {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static DateTimeFormatter dateFormat =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss.SSS").withZone(ZoneId.systemDefault());
+
+  private static final int CONNECT_TIMEOUT = 30 * 1000;
+  private static final int READ_TIMEOUT = 30 * 1000;
```

Comment:
should we re-use the existing timeout values instead ? If not atleast we should make this configurable

Reply:
Wont be application once I use the google client instead.